docs/developer/adding-community-components.md

```diff
@@ -0,0 +1,53 @@
+# Adding community components
+
+[Community components][cc] are components that are implemented and maintained by the community.
+
+## Before opening a proposal
+
+The first step is to ensure that the proposal meets the following criteria and does not duplicate existing proposals:
+
+* Avoid overlapping functionalities.
+* Avoid components that can be implemented as [modules][module].
+* Avoid components that affect our dependencies in an undesired way, such as pulling in an incompatible version or bloating the collector.
+* Make sure that the code licenses are compatible with Alloy's [license][].
+
+While not mandatory, it is beneficial if:
+
+* The component comes from [Opentelemetry's contrib repository][otel].
+* The component supports all the [platforms that Alloy supports][platforms].
+
```

Comment:
Should we talk about how "community components" is being rolled out slowly and we'll initially be slow to accept new components, so that we can refine the process and not get overwhelmed by dozens of new components?